File: SrdDatabase/ClientApp/src/models/event.ts
import { PagedParameters, PagedResults } from "./shared";

export interface Event {
    id?: number;
    eventTypeId?: number;
    eventType?: string;
    congregationId?: number;
    congregation?: string;
    parishId?: number;
    parish?: string;
    archdeaconryId?: number;
    archdeaconry?: string;
    description?: string;
    firstPersonName?: string;
    secondPersonName?: string;
    personNames?: string[];
    date?: Date;
    createdBy?: number;
}

export interface EventToSend extends Omit<Event, "date"> {
    date?: string;
}

export interface EventType {
    id: number;
    name: string;
    associatedWithParish: boolean;
    involvesDescription: boolean;
    involvesFirstPerson: boolean;
    involvesSecondPerson: boolean;
}

export interface EventDetails {
    event: Event;
}

export interface EventParameters extends PagedParameters {
    eventTypeId?: number;
    personName?: string;
    description?: string;
    parishId?: number;
    archdeaconryId?: number;
    congregationId?: number;
    startDate?: Date;
    endDate?: Date;
}

export interface EventParametersToSend extends Omit<EventParameters, "startDate" | "endDate"> {
    startDate?: string;
    endDate?: string;
}

export interface EventResults extends PagedResults {
    events: Event[];
}

File: SrdDatabase/Models/Events/Event.cs
using System;

namespace SrdDatabase.Models.Events
{
    public class Event
    {
        public int Id { get; }

        public sbyte EventTypeId { get; }

        public string EventType { get; }

        public int? CongregationId { get; }

        public string Congregation { get; }

        public int ParishId { get; }

        public string Parish { get; }

        public int ArchdeaconryId { get; }

        public string Archdeaconry { get; }

        public string FirstPersonName { get; }

        public string SecondPersonName { get; }

        public string Description { get; }

        public DateTime Date { get; }

        public int CreatedBy { get; }

        public Event(
            int id,
            sbyte eventTypeId,
            string eventType,
            int? congregationId,
            string congregation,
            int parishId,
            string parish,
            int archdeaconryId,
            string archdeaconry,
            string firstPersonName,
            string secondPersonName,
            string description,
            DateTime date,
            int createdBy)
        {
            Id = id;
            EventTypeId = eventTypeId;
            EventType = eventType;
            CongregationId = congregationId;
            Congregation = congregation;
            ParishId = parishId;
            Parish = parish;
            ArchdeaconryId = archdeaconryId;
            Archdeaconry = archdeaconry;
            FirstPersonName = firstPersonName;
            SecondPersonName = secondPersonName;
            Description = description;
            Date = date;
            CreatedBy = createdBy;
        }

        // for Dapper
        public Event()
        {
        }
    }
}
